src/AdapterAbstract.php

```
@@ -0,0 +1,633 @@
+   */
+  public function fetchColumn( $sql, array $parameters = null, $master = false ) {
+
+    $statement = $this->query( $sql, $parameters, $master );
+
+    return ( $statement->columnCount() === 0 )
+           ? []
+           : $statement->fetchAll( \PDO::FETCH_COLUMN );
+
+  } // fetchColumn
+
+
+  /**
+   * Provides compatibility with ZF1's fetchAssoc, an assoc array keyed by the first column
+   *
+   * @param string $sql
+   * @param array  $parameters
+   * @param bool   $master
+   *
+   * @return array
+   */
+  public function fetchAll( $sql, array $parameters = null, $master = false ) {
+
+    $statement = $this->query( $sql, $parameters, $master );
+
+    return ( $statement->columnCount() === 0 )
+           ? []
+           : $statement->fetchAll( \PDO::FETCH_ASSOC );
+
+  } // fetchAll
+
+
+  /**
+   * @param string $sql
+   * @param array  $parameters
+   * @param bool   $master
+   *
+   * @return array
+   */
+  public function fetchAssoc( $sql, array $parameters = null, $master = false ) {
+
+    $statement = $this->query( $sql, $parameters, $master );
+    $results   = [];
+
+    while ( $row = $statement->fetch( \PDO::FETCH_ASSOC ) ) {
+
+      // Retrieves the first associative value from the array
+      $values = array_values( array_slice( $row, 0, 1 ) );
+
+      $results[ $values[ 0 ] ] = $row;
+
+    } // while fetch
+
+    return $results;
+
+  } // fetchAssoc
+
+
+  /**
+   * @throws Behance\NBD\Dbal\Exceptions\QueryRequirementException when less than 2 columns are selected
+   *
+   * @param string $sql
+   * @param array  $parameters
+   * @param bool   $master
+   *
+   * @return array
+   */
+  public function fetchPairs( $sql, array $parameters = null, $master = false ) {
+
+    $statement = $this->query( $sql, $parameters, $master );
+
+    // Columns will be 0 on an empty result set, which is not a violation
+    if ( $statement->columnCount() === 0 ) {
+      return [];
+    }
+
+    if ( $statement->columnCount() < 2 ) {
```

Would we want to throw an exception if it's != 2? If I do `fetchPairs` on `SELECT *` or `SELECT a,b,c` that seems like a misuse.